### PR TITLE
CASSANDRA-16150 Upgrade to a snakeyaml version without CVE

### PR DESCRIPTION
**Pull request by crazylab**
Fix for CVE-2017-18640: https://nvd.nist.gov/vuln/detail/CVE-2017-18640 has been patched into snakeyaml 1.26

### CODE REVIEW

**Comment by crazylab** [comment]
@dcapwell can you please have a look?

**Comment by dcapwell** [comment]
what JIRA is this related to?  We currently use 1.23.

**Comment by dcapwell** [comment]
@crazylab is there a JIRA for this, if not can you create one?

**Comment by crazylab** [comment]
@dcapwell created a card https://issues.apache.org/jira/browse/CASSANDRA-16150

File: build.xml [hunk 1]
```diff
@@ -583,8 +583,7 @@
           <dependency groupId="com.googlecode.json-simple" artifactId="json-simple" version="1.1"/>
           <dependency groupId="com.boundary" artifactId="high-scale-lib" version="1.0.6"/>
           <dependency groupId="com.github.jbellis" artifactId="jamm" version="${jamm.version}"/>
-
-          <dependency groupId="org.yaml" artifactId="snakeyaml" version="1.23"/>
+          <dependency groupId="org.yaml" artifactId="snakeyaml" version="1.26"/>
```
Comment (dcapwell) [inline build.xml:586]: sadly this isn't enough as we commit the binary into the repo: see https://github.com/apache/cassandra/tree/trunk/lib

Comment (crazylab) [reply build.xml:586]: Added in another commit to add the library.

Comment (crazylab) [reply build.xml:586]: Just for my understanding, why we are maintaining libraries in the codebase?

Comment (dcapwell) [reply build.xml:586]: I don't know, it was this way before I joined.  For whatever reason test libs are pulled from maven central but src libs are not and pulled from local disk.

**Pull request closed without merging**